src/poliastro/twobody/propagation.py

```diff
@@ -103,6 +106,60 @@ def cowell(k, r0, v0, tof, rtol=1e-10, *, ad=None, callback=None, nsteps=1000):
     return r, v
 
 
+def mean_motion(k, r0, v0, tof, **kwargs):
+    """Propagates orbit using mean motion
+
+    Parameters
+    ----------
+    k : float
+        Gravitational constant of main attractor (km^3 / s^2).
+    r0 : array
+        Initial position (km).
+    v0 : array
+        Initial velocity (km).
+    ad : function(t0, u, k), optional
+         Non Keplerian acceleration (km/s2), default to None.
+    tof : float
+        Time of flight (s).
+
+    Note
+    -----
+    This method takes initial \vec{r}, \vec{v}, calculates classical orbit parameters,
+    increases mean anomaly and performs inverse transformation to get final \vec{r}, \vec{v}
+    The logic is based on formulae (4), (6) and (7) from http://dx.doi.org/10.1007/s10569-013-9476-9
+    """
+
+    # get the initial true anomaly and orbit parameters that are constant over time
+    p, ecc, inc, raan, argp, nu0 = rv2coe(k, r0, v0)
+    # get the initial mean anomaly
+    M0 = nu_to_M(nu0, ecc)
+    # elliptic or hyperbolic orbits
+    if ecc != 1.0:
```

Comment:
Just for curiosity: what is the asymptotic behavior of this when the eccentricity is very close to 1.0? We should add a `# TODO: ` here to remind us that this needs better validation. Perhaps in the original paper there's some information.